File: android2/app/src/main/java/com/xbot/xbot/model/Expression.java
package com.xbot.xbot.model;

/** Seven common expressions (Ekman six + neutral). */
public enum Expression {
    NEUTRAL,
    HAPPY,
    SAD,
    SURPRISED,
    ANGRY,
    DISGUSTED,
    FEARFUL;

    public String getLabel() {
        switch (this) {
            case NEUTRAL: return "中性";
            case HAPPY: return "高兴";
            case SAD: return "伤心";
            case SURPRISED: return "惊讶";
            case ANGRY: return "愤怒";
            case DISGUSTED: return "厌恶";
            case FEARFUL: return "恐惧";
            default: return name();
        }
    }

    public String getApiKey() {
        return name().toLowerCase();
    }

    public String getEmoji() {
        switch (this) {
            case NEUTRAL: return "😐";
            case HAPPY: return "😄";
            case SAD: return "😢";
            case SURPRISED: return "😲";
            case ANGRY: return "😠";
            case DISGUSTED: return "🤢";
            case FEARFUL: return "😨";
            default: return "";
        }
    }
}

File: xbot-native/app/src/main/java/com/xbot/android/model/Expression.kt
package com.xbot.android.model

import androidx.compose.ui.graphics.Color

/**
 * 7 种常见表情（Ekman 经典 6 种 + 中性）。对应 Flutter Expression 枚举。
 */
enum class Expression(val apiName: String, val label: String, val emoji: String) {
    NEUTRAL("neutral", "中性", "😐"),
    HAPPY("happy", "高兴", "😄"),
    SAD("sad", "伤心", "😢"),
    SURPRISED("surprise", "惊讶", "😲"),
    ANGRY("angry", "愤怒", "😠"),
    DISGUSTED("disgust", "厌恶", "🤢"),
    FEARFUL("fear", "恐惧", "😨");

    companion object {
        /** 后端 7 类 key → 枚举。 */
        fun fromApiName(key: String?): Expression =
            entries.firstOrNull { it.apiName == key } ?: NEUTRAL
    }
}

/** 表情识别结果：最可能的表情、其置信度、以及全部表情的打分（调试用）。 */
data class ExpressionResult(
    val expression: Expression,
    val score: Float,
    val scores: Map<Expression, Float>,
) {
    companion object {
        val NEUTRAL = ExpressionResult(
            expression = Expression.NEUTRAL,
            score = 1f,
            scores = mapOf(Expression.NEUTRAL to 1f),
        )
    }
}
